tests/test_rotation.py

```diff
@@ -0,0 +1,52 @@
+from unittest import TestCase
+
+import numpy as np
+from scipy.spatial.transform import Rotation as sp_rot
+
+from aspire.utils.rotation import Rotation
+from aspire.utils.types import utest_tolerance
+
+
+class UtilsTestCase(TestCase):
+    def setUp(self):
+        self.rot_obj = Rotation(32, seed=0, dtype=np.float32)
+
+    def testRotMatrices(self):
+        rots_ref = (
+            sp_rot.from_euler("ZYZ", self.rot_obj.angles, degrees=True)
+            .as_matrix()
+            .astype(self.rot_obj.dtype)
+        )
+        self.assertTrue(np.allclose(self.rot_obj.rot_matrices, rots_ref))
+
+    def testRotAngles(self):
+        rot = sp_rot.from_matrix(self.rot_obj.rot_matrices)
+        angles = rot.as_euler(self.rot_obj.rot_seq, degrees=True).astype(
+            self.rot_obj.dtype
+        )
+        self.assertTrue(np.allclose(self.rot_obj.angles, angles))
+
+    def testTranspose(self):
+        rot_mat = self.rot_obj.rot_matrices
+        rot_mat_t = self.rot_obj.T
+        self.assertTrue(np.allclose(rot_mat_t, np.transpose(rot_mat, (0, 2, 1))))
+
+    def testMultiplication(self):
+        rot_obj_t = Rotation(32, seed=0)
+        rot_obj_t.rot_matrices = self.rot_obj.T
+        result = self.rot_obj * rot_obj_t
+        for i in range(self.rot_obj.num_rots):
+            self.assertTrue(
+                np.allclose(
+                    np.eye(3), result[i], atol=utest_tolerance(self.rot_obj.dtype)
+                )
+            )
+
+    def testRegisterRots(self):
+        q_ang = [[45, 45, 45]]
```

Comment:
Pick some random angles.

Reply:
Done in new commit e863d01.

Reply:
Why not use `uniform_random_angles` for this?

Reply:
Changed in new commit 2602b07.